engine/lite/request-response/components-api/src/main/scala/pl/touk/nussknacker/engine/requestresponse/api/RequestResponseSourceFactory.scala

```diff
@@ -33,8 +46,50 @@ trait RequestResponseSource[T] extends BaseLiteSource[Any] {
 
   private val contextInitializer = new BasicContextInitializer[Any](Unknown)
 
-  override def transform(record: Any): ContextVariables = {
-    contextInitializer.convertToInitialVariables(record)
+  // TODO: Consider moving it to BaseLiteSource
+  override def createTransformation[F[_]: Monad](
+      componentContext: customComponentTypes.CustomComponentContext[F]
+  ): Any => ValidatedNel[ErrorType, Context] =
+    request => {
+      val context = Context(contextIdGenerator.nextContextId())
+
+      Validated
+        .fromEither(Try((transform(request), extractTraceId(request))).toEither)
+        .map { case (contextVariables, traceId) =>
+          val contextWithVariables = context.withVariables(contextVariables.variables)
+
+          traceId
+            .map(contextWithVariables.withTraceId)
+            .getOrElse(contextWithVariables)
+        }
+        .leftMap(ex =>
+          NuExceptionInfo(
+            Some(NodeComponentInfo(componentContext.nodeId, ComponentType.Source, "unknown")),
+            ex,
+            context
+          )
+        )
+        .toValidatedNel
+    }
+
+  // protected for overriding purposes
+  protected def extractTraceId(input: Any): Option[TraceId] =
+    input match {
```

Comment:
This pattern matching is kind of hack.. Maybe we should try to pass headers as additional param: 

```
(Any, Map[String, String]) => ValidatedNel[ErrorType, Context]
```
??